sidecar/src/self_telemetry.rs

```diff
@@ -65,43 +69,79 @@ impl<'a> MetricData<'a> {
             futures.push(self.send(
                 self.logs_created,
                 count as f64,
-                vec![Tag::new("level", level.as_str().to_lowercase()).unwrap()],
+                vec![
+                    Tag::new("level", level.as_str().to_lowercase()).unwrap(),
```

Comment:
I would be remiss if I didn't call out all of these unwraps on `Tag::new()`. If the `Result` is not Ok, should we log an error and not attempt to push the future? I assume we could wrap that logic in a helper function.

Reply:
Actually, [this](https://github.com/DataDog/libdatadog/pull/450#issuecomment-2130221322) would solve the problem. So feel free to leave the unwraps for now.